AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -357,6 +382,14 @@ - (void)performBatchUpdates:(void (^)())updates completion:(void (^)(BOOL))compl
   [self performBatchAnimated:YES updates:updates completion:completion];
 }
 
+- (void)registerSupplementaryNodeOfKind:(NSString *)elementKind
+{
+  ASDisplayNodeAssert(elementKind != nil, @"A kind is needed for supplementary node registration");
+  [_registeredSupplementaryKinds addObject:elementKind];
+  [self registerClass:[UICollectionReusableView class] forSupplementaryViewOfKind:elementKind
```

Comment:
I don't see one for `UICollectionReusableView`, but you are correct there is a `_ASTableViewCell` subclass. It looks like it's being used to bubble relayout events out of individual cells. Great idea. Not sure how valuable it is to add an `_ASCollectionReusableView` as of right now. I would like to think about it a bit more post this diff, since a supplementary view's bounds is often defined by the layout outside of the node's scope.